test/integration/associations/scope.test.js

```diff
@@ -365,12 +359,16 @@ describe(Support.getTestDialectTeaser('associations'), () => {
             await Promise.all([
               postA0.addCategory(categoryA),
               postA0.createTag(),
-              postB0.setCategories([categoryB]),
               postB0.addTag(tagA),
               postC0.createCategory(),
-              postC0.setTags([tagB]),
             ]);
 
+            // we're calling 'setX' methods after the different 'addX' methods because
+            // setCategories is not supposed to overwrite tags and vice-versa.
+            // tags & categories use the same through table so this could happen is the association scope is not handled correctly.
+            await postB0.setCategories([categoryB]);
+            await postC0.setTags([tagB]);
+
             const [postACategories, postBCategories, postCCategories, postATags, postBTags, postCTags] = await Promise.all([
               this.postA.getCategories(),
               this.postB.getCategories(),
```

Comment:
GitHub doesn't let me comment on line 381-382, but can you delete that comment?

Reply:
Done!